### DISCUSSION TITLE
How do you load an OBJ file that has several texture files associated in the MTL file ?

### DISCUSSION THREAD

**Discussion by segobytes** [pyvista/pyvista#4763] [Q&A] [answered]
Hello all, 
I am attaching for reference a zip file containing an OBJ file, an MTL file and 3 texture jpg files.
The MTL file describes the 3 texture files as well as a metallic finish that has no associated image file.
There was a solution documented in https://github.com/pyvista/pyvista/discussions/2622 that worked with version 0.39, but it does not work anymore in v0.41
In the context of pyvista 0.41,  is there a way to load this OBJ files including all the textures described in the MTL file ?     
[Spaceship.zip](https://github.com/pyvista/pyvista/files/12328464/Spaceship.zip)

**Comment by segobytes** [comment]
So I found a way by importing with vtk and exporting a vtp file with a single png texture file,  but I'd like to find a way that does not require to export the vtp & png file if possible... Anyone ?

```python
importer = vtk.vtkOBJImporter()
importer.SetFileName(obj_file_path)
importer.SetFileNameMTL(mtl_file_path)
importer.SetTexturePath(tex_directory_path)
importer.Update()

exporter = vtk.vtkSingleVTPExporter()
exporter.SetRenderWindow(importer.GetRenderWindow())
exporter.SetFilePrefix("export_path_and_file_prefix")
exporter.Write()
# this will write 2 files (vtp and png), let's call them vtp_path and tex_path

mesh = pv.read(vtp_path)
tex = pv.read_texture(tex_path)
plotter = pv.Plotter(notebook=False)
plotter.add_mesh(mesh, texture=tex)
plotter.show()
```

**Comment by segobytes** [comment]
Found the way...
```python
importer = vtk.vtkOBJImporter()
importer.SetFileName(obj_path)
importer.SetFileNameMTL(mtl_path)
importer.SetTexturePath(tex_path)
importer.SetRenderWindow(self.plotter.render_window)
importer.Update()
```

**Reply by songhori** [reply]
Hi segobytes
That was a great solution.
I was seeking it for a long time.
Thanks so much

**Reply by tkoyama010** [reply, marked as answer]
Thanks for this discussion. I added new method to Plotter class in #6265. I hope it helps you. 
```python
>>> from pathlib import Path
>>> filename = examples.download_doorman(load=False)
>>> filename_mtl = str(Path(filename).with_suffix('.mtl'))
>>> texture_path = str(Path(filename).parent)
>>> pl = pv.Plotter()
>>> pl.import_obj(
...     filename,
...     filename_mtl=filename_mtl,
...     texture_path=texture_path,
... )
>>> pl.show(cpos="xy")
```
![image](https://github.com/pyvista/pyvista/assets/7513610/6f901cea-6071-4777-ad10-b079bceadcf0)
